src/util.cc

```diff
@@ -825,6 +825,36 @@ double GetLoadAverage() {
 }
 #else
 double GetLoadAverage() {
+# if defined(__linux__)
+  // Only try to open the file once.
+  static int fd = -1;
```

Comment:
what is the reason to make the variable static?

Reply:
This reduces the number of system calls per call to this function: Rather than open(), read(), close() every time, we open() once and then read() each following time.  Caching the file descriptor seemed appropriate in this case.

Reply:
Oh I see. This is a cool trick. I am fine with this